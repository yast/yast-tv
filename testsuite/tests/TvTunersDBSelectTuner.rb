# encoding: utf-8

module Yast
  class TvTunersDBSelectTunerClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"

      Tv.tuners_database = nil
      @tuner = Tv.TunersDBSelectTuner("bttv", 1)
      DUMP(@tuner)


      Tv.tuners_database = {}
      @tuner = Tv.TunersDBSelectTuner("bttv", 1)
      DUMP(@tuner)

      Tv.tuners_database = {
        "bttv" => [
          { "name" => "No tuner", "parameters" => { "tuner" => "4" } },
          { "name" => "Alps HSBH1", "parameters" => { "tuner" => "9" } }
        ]
      }
      @tuner = Tv.TunersDBSelectTuner("bttv", 2)
      DUMP(@tuner)

      @tuner = Tv.TunersDBSelectTuner("bttv", 3)
      DUMP(@tuner)

      nil
    end
  end
end

Yast::TvTunersDBSelectTunerClient.new.main
