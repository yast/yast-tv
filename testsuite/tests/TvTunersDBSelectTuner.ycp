{

    include "testsuite.ycp";

    map READ = $[ "target" : $[ "size" : -1 ] ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

	Tv::tuners_database = nil;
	map tuner = Tv::TunersDBSelectTuner("bttv", 1);
	DUMP(tuner);


	Tv::tuners_database = $[];
	tuner = Tv::TunersDBSelectTuner("bttv", 1);
	DUMP(tuner);

	Tv::tuners_database = $["bttv" :
		[ $["name":"No tuner",
			"parameters":$["tuner":"4"]],
		  $["name":"Alps HSBH1",
		    "parameters":$["tuner":"9"]]]
	];
	tuner = Tv::TunersDBSelectTuner("bttv", 2);
	DUMP(tuner);

	tuner = Tv::TunersDBSelectTuner("bttv", 3);
	DUMP(tuner);
}
