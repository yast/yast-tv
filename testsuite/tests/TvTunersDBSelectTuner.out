Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp"
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system nil
Dump	$[]
Dump	$[]
Dump	$["name":"Alps HSBH1", "parameters":$["tuner":"9"]]
Dump	$[]
